resources/github-actions.md

```diff
@@ -176,51 +176,144 @@ The example code below demonstrates building and testing with conda packages fro
+# make sure they can be found locally
+conda config --system --add channels "${LIBKVIKIO_CHANNEL}"
+conda config --system --add channels "${LIBRMM_CHANNEL}"
+conda config --system --add channels "${RMM_CHANNEL}"
+```
+
+Then copy the following into every script in the `ci/` directory that is doing `conda` installs.
 
-rapids-mamba-retry install \
-  --channel "${CPP_CHANNEL}" \
-  --channel "${PYTHON_CHANNEL}" \
-  --channel "${LIBRMM_CHANNEL}" \
-  --channel "${LIBKVIKIO_CHANNEL}" \
-  --channel "${RMM_CHANNEL}" \
-  cudf libcudf
+```shell
+source ./ci/use_conda_packages_from_prs.sh
 ```
 
-Note that the custom channel for PR artifacts is needed in the build scripts _and_ the test scripts, for C++ _and_ Python.
-If building/testing a Python package that depends on a C++ library, it is necessary to use PR artifacts from that C++ library and not just Python (e.g. if testing `rmm` artifacts, you must use the corresponding `librmm` CI artifacts as well as `rmm`).
-In some repos, the `test_python.sh` is quite complicated with multiple calls to conda/mamba.
-We recommend that the Python and C++ artifact channels should be added to every call of `rapids-mamba-retry` / `rapids-conda-retry` "just in case."
+**Note:** By default `rapids-get-pr-conda-artifact` uses the most recent commit from the specified PR.
+A commit hash from the dependent PR can be added as an optional 4th argument to pin testing to a specific commit.
+
+## Using Wheel CI Artifacts in Other PRs
+
+To use wheels produced by other PRs' CI:
+
+* download the wheels at the beginning of CI jobs
+* constrain `pip` to always use them
+
+Consider the following examples.
+
+**Example:** Building `libcuml` (C++) using `librmm` and `libraft` PR artifacts.
+
+Add a new file called `ci/use_wheels_from_prs.sh`.
+
+```shell
+# ci/use_wheels_from_prs.sh
+
+RAPIDS_PY_CUDA_SUFFIX=$(rapids-wheel-ctk-name-gen "${RAPIDS_CUDA_VERSION}")
+
+# download wheels, store the directories holding them in variables
+LIBRMM_WHEELHOUSE=$(
+  RAPIDS_PY_WHEEL_NAME="librmm_${RAPIDS_PY_CUDA_SUFFIX}" rapids-get-pr-wheel-artifact rmm 1678 cpp
+)
+LIBRAFT_WHEELHOUSE=$(
+  RAPIDS_PY_WHEEL_NAME="libraft_${RAPIDS_PY_CUDA_SUFFIX}" rapids-get-pr-wheel-artifact raft 2433 cpp
+)
+
+# write a pip constraints file saying e.g. "whenever you encounter a requirement for 'librmm-cu12', use this wheel"
+cat > /tmp/constraints.txt <<EOF
+librmm-${RAPIDS_PY_CUDA_SUFFIX} @ file://$(echo ${LIBRMM_WHEELHOUSE}/librmm_*.whl)
+libraft-${RAPIDS_PY_CUDA_SUFFIX} @ file://$(echo ${LIBRAFT_WHEELHOUSE}/libraft_*.whl)
+EOF
+
+export PIP_CONSTRAINT=/tmp/constraints.txt
+```
+
+Then copy the following into every script in the `ci/` directory that is doing `pip` installs or wheels builds with e.g. `pip wheel`.
+
+```shell
+source ./ci/use_wheels_from_prs.sh
+```
+
+This should generally be enough.
+If any of the other CI scripts are already setting environment variable `PIP_CONSTRAINT`, you may need to
```

Comment:
```suggestion
If any of the other CI scripts are already setting the environment variable `PIP_CONSTRAINT`, you may need to
```

Reply:
I don't think the form without "the" is grammatically incorrect, but I'll commit this because I can't think of any reason to choose one form over the other.